options.go

```diff
@@ -31,6 +36,10 @@ type dialerConfig struct {
 	sqladminOpts   []apiopt.ClientOption
 	dialOpts       []DialOption
 	refreshTimeout time.Duration
+	useIAMAuth     bool
```

Comment:
nit: it's probably more correct to use `useIAMAuthN` here as well, since technically AuthZ is always used

Reply:
Definitely. Done.